src/nvim/eval.c

```diff
@@ -8021,6 +8022,40 @@ static void f_call(typval_T *argvars, typval_T *rettv)
   (void)func_call(func, &argvars[1], selfdict, rettv);
 }
 
+// "capture(command)" function
+static void f_capture(typval_T *argvars, typval_T *rettv)
+{
+    int save_msg_silent = msg_silent;
+    garray_T *save_capture_ga = capture_ga;
+
+    if (check_secure()) {
+      return;
+    }
+
+    capture_ga = xmalloc(sizeof(garray_T));
+    ga_init(capture_ga, (int)sizeof(char), 80);
+
+    msg_silent++;
+    if (argvars[0].v_type == VAR_LIST) {
+      listitem_T *li;
+      for (li = argvars[0].vval.v_list->lv_first;
+           li != NULL; li = li->li_next) {
+        do_cmdline_cmd((char *)get_tv_string(&li->li_tv));
+      }
+    } else {
+      do_cmdline_cmd((char *)get_tv_string(&argvars[0]));
+    }
+    msg_silent = save_msg_silent;
+
+    ga_append(capture_ga, NUL);
+    rettv->v_type = VAR_STRING;
+    rettv->vval.v_string = vim_strsave((char_u *)capture_ga->ga_data);
```

Comment:
Do not copy string here. This is absolutely useless action, you *already* have an allocated string, just use it. And remove `ga_clear`.